gst/constants.go

```diff
@@ -709,11 +709,11 @@ type ProgressType int
 
 // Type castings of ProgressTypes
 const (
-	ProgressTypeStart     ProgressType = C.GST_PROGRESS_TYPE_START    // (0) – A new task started.
-	ProgressTypeContinue  ProgressType = C.GST_PROGRESS_TYPE_CONTINUE // (1) – A task completed and a new one continues.
-	ProgressTypeComplete  ProgressType = C.GST_PROGRESS_TYPE_COMPLETE // (2) – A task completed.
-	ProgressTypeCancelled ProgressType = C.GST_PROGRESS_TYPE_CANCELED // (3) – A task was canceled.
-	ProgressTypeError     ProgressType = C.GST_PROGRESS_TYPE_ERROR    // (4) – A task caused an error. An error message is also posted on the bus.
+	ProgressTypeStart    ProgressType = C.GST_PROGRESS_TYPE_START    // (0) – A new task started.
+	ProgressTypeContinue ProgressType = C.GST_PROGRESS_TYPE_CONTINUE // (1) – A task completed and a new one continues.
+	ProgressTypeComplete ProgressType = C.GST_PROGRESS_TYPE_COMPLETE // (2) – A task completed.
+	ProgressTypeCanceled ProgressType = C.GST_PROGRESS_TYPE_CANCELED // (3) – A task was canceled.
```

Comment:
because of this line this PR is a breaking change

Reply:
Hmmm cancelled is double l in British English... and it's how I would write it... is one l American English?

Reply:
https://www.grammarly.com/blog/canceled-vs-cancelled/

Yup

If we're choosing American then fine!

Reply:
I fixed the spellings in this branch based on the errors reported in #35 

Most of the spellings in this repo are US (misspell reported a lot more issues when using UK spelling)

But I discovered that gstreamer itself uses the US spelling (seen in `C.GST_PROGRESS_TYPE_CANCELED` for example), but `glib` however uses british spelling e.g. [`C.g_cancellable_set_error_if_cancelled`](https://github.com/go-gst/go-glib/blob/main/glib/gcancellable.go#L53)